From fad0b3df91b92bba2acd369010e166023f4ca970 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Thu, 15 Dec 2011 15:49:36 +0800
Subject: [PATCH 250/449] ENGR00170343 unit test: update v4l2 output autotest

update v4l2 output autotest.

for full test mode, please export FULLTEST=1 then run autorun-v4l2.sh
Note: it will take a long time.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 test-utils.sh                        |  1 +
 test/mxc_v4l2_test/autorun-v4l2.sh   | 76 ++++++++++++++++++++++++++++++++++++
 test/mxc_v4l2_test/mxc_v4l2_output.c |  2 +-
 3 files changed, 78 insertions(+), 1 deletion(-)

diff --git a/test-utils.sh b/test-utils.sh
index d8ce686..e845d5f 100755
--- a/test-utils.sh
+++ b/test-utils.sh
@@ -174,6 +174,7 @@ platform()
 		MX51* )       		plat=IMX51 ;;
 		MX53* )       		plat=IMX53 ;;
 		MX50* )       		plat=IMX50 ;;
+		*MX6* )       		plat=IMX6 ;;
 
 		* )     ;;
 	esac
diff --git a/test/mxc_v4l2_test/autorun-v4l2.sh b/test/mxc_v4l2_test/autorun-v4l2.sh
index eb7822e..798f86e 100755
--- a/test/mxc_v4l2_test/autorun-v4l2.sh
+++ b/test/mxc_v4l2_test/autorun-v4l2.sh
@@ -2,6 +2,9 @@
 
 source /unit_tests/test-utils.sh
 
+start_time=`date`
+echo =============== test start from $start_time ==============================
+
 #
 # Exit status is 0 for PASS, nonzero for FAIL
 #
@@ -29,6 +32,11 @@ if [ "$(platform)" != IMX25_3STACK ]; then
 check_devnode "/dev/video16"
 fi
 
+if ([ "$(platform)" = IMX51 ] || [ "$(platform)" = IMX53 ] \
+	|| [ "$(platform)" = IMX6 ]); then
+check_devnode "/dev/video17"
+fi
+
 # Turn off fb blanking
 echo -e "\033[9;0]" > /dev/tty0
 
@@ -37,9 +45,72 @@ DISPLAY=0;
 else
 DISPLAY=3;
 fi
+
+if ([ "$(platform)" = IMX51 ] || [ "$(platform)" = IMX53 ] \
+	|| [ "$(platform)" = IMX6 ]); then
+DISPLAY=/dev/video17
+fi
+
 #
 # V4L2 Output Tests
 #
+
+if ([ "$(platform)" = IMX51 ] || [ "$(platform)" = IMX53 ] \
+	|| [ "$(platform)" = IMX6 ]); then
+DISPW=`cat /sys/class/graphics/fb0/mode | awk -F ':' '{print $2}' | awk -F 'x' '{print $1}'`
+DISPH=`cat /sys/class/graphics/fb0/mode | awk -F ':' '{print $2}' | awk -F 'x' '{print $2}' | awk -F 'p' '{print $1}'`
+DISPW_DIV_2=`expr $DISPW / 2`
+DISPW_DIV_8=`expr $DISPW / 8`
+DISPH_DIV_2=`expr $DISPH / 2`
+DISPH_DIV_8=`expr $DISPH / 8`
+
+if [ "$FULLTEST" = '1' ]; then
+IWCASE="$DISPW_DIV_8 $DISPW_DIV_2 $DISPW"
+IHCASE="$DISPH_DIV_8 $DISPH_DIV_2 $DISPH"
+OWCASE="$DISPW_DIV_8 $DISPW_DIV_2 $DISPW"
+OHCASE="$DISPH_DIV_8 $DISPH_DIV_2 $DISPH"
+else
+IWCASE="$DISPW"
+IHCASE="$DISPH"
+OWCASE="$DISPW"
+OHCASE="$DISPH"
+fi
+
+echo =========== do test on display $DISPW $DISPH ====================
+for IW in $IWCASE; do
+for IH in $IHCASE; do
+for OW in $OWCASE; do
+for OH in $OHCASE; do
+for ROT in 0 90; do
+for VF in 0 1; do
+for HF in 0 1; do
+	ICW=`expr $IW / 4`
+	ICH=`expr $IH / 4`
+	OCW=`expr $OW / 4`
+	OCH=`expr $OH / 4`
+	# resizing, rotation, flip test
+	run_testcase "./mxc_v4l2_output.out -iw $IW -ih $IH -ow $OW -oh $OH -d $DISPLAY -r $ROT -vf $VF -hf $HF -fr 60"
+	# resizing, rotation, flip with crop test
+	run_testcase "./mxc_v4l2_output.out -iw $IW -ih $IH -cr $ICW $ICH $ICW $ICH 0 -ow $OW -oh $OH -ol $OCW -ot $OCH -d $DISPLAY -r $ROT -vf $VF -hf $HF -fr 60"
+	# resizing, rotation, flip test for deinterlacing
+	run_testcase "./mxc_v4l2_output.out -iw $IW -ih $IH -ow $OW -oh $OH -d $DISPLAY -r $ROT -vf $VF -hf $HF -v 0 -fr 60"
+	run_testcase "./mxc_v4l2_output.out -iw $IW -ih $IH -ow $OW -oh $OH -d $DISPLAY -r $ROT -vf $VF -hf $HF -v 2 -fr 60"
+	# resizing, rotation, flip with crop test for deinterlacing
+	run_testcase "./mxc_v4l2_output.out -iw $IW -ih $IH -cr $ICW $ICH $ICW $ICH 0 -ow $OW -oh $OH -ol $OCW -ot $OCH -d $DISPLAY -r $ROT -vf $VF -hf $HF -v 0 -fr 60"
+done
+done
+done
+done
+done
+done
+done
+
+# user pointer test
+run_testcase "./mxc_v4l2_output.out -iw $DISPW_DIV_2 -ih $DISPH_DIV_2 -ow $DISPW -oh $DISPH -d $DISPLAY -u"
+run_testcase "./mxc_v4l2_output.out -iw $DISPW_DIV_2 -ih $DISPH_DIV_2 -ow $DISPW_DIV_2 -oh $DISPH_DIV_2 -d $DISPLAY -u"
+
+else
+
 # SDC output size test cases
 if [ "$(platform)" != IMX25_3STACK ]; then
 for SIZE in 32 64 80 96 112 128 144 160 176 192 208 224 240; do
@@ -64,6 +135,7 @@ else
 	run_testcase "./mxc_v4l2_output.out -iw 720 -ih 512 -ow 240 -oh 184 -d $DISPLAY -fr 60"
 fi
 fi
+fi
 
 if [ $TEST_CAMERA = 1 ]; then
 
@@ -87,4 +159,8 @@ fi
 
 
 print_status
+
+stop_time=`date`
+echo =============== test start from $stop_time ==============================
+
 exit $STATUS
diff --git a/test/mxc_v4l2_test/mxc_v4l2_output.c b/test/mxc_v4l2_test/mxc_v4l2_output.c
index 19f9291..8c3517e 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_output.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_output.c
@@ -316,7 +316,7 @@ mxc_v4l_output_test(FILE *in)
 					}
 					break;
 				}
-				count = 100;
+				count = 30;
 			}
                         gen_fill_pattern(buffers[buf.index].start, i);
                 }
-- 
1.8.0

